syntax = "proto3";

package messages;

option go_package = "./messages";  // Altere este caminho conforme necess√°rio

// Mensagens do cliente para o Broker
message ClientMessage {
    string request = 1; // Por exemplo, "GET_DEVICE_STATE"
}

message ClienteResponse {
    string response = 1; // Resposta do Broker para o cliente
}

// Mensagens do dispositivo para o Broker
message DeviceMessage {
    string device_id = 1;
    string data = 2;
}

message DeviceResponse {
    string device_id = 1;
    string response = 2;
}

// Mensagens de descoberta
message DiscoverMessage {
    string request = 1; // Por exemplo, "DISCOVER_DEVICES"
}

message DiscoverResponse {
    repeated Device device = 1;

    message Device {
        string device_id = 1;
        string ip = 2;
        int32 port = 3;
    }
}